opentelemetry/src/sdk/trace/span.rs

```diff
@@ -70,6 +70,21 @@ impl Span {
     {
         self.data.as_mut().map(f)
     }
+
+    /// Convert information in this span into `exporter::trace::SpanData`.
+    /// This function copies all data from the current span, which will create a
+    /// overhead.
+    pub fn exported_data(&self) -> Option<crate::sdk::export::trace::SpanData> {
```

Comment:
Hm I wonder if a kind of `ReadableSpan` makes sense here https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/sdk.md#additional-span-interfaces

Reply:
I guess it makes sense. But on the other hand, `SpanData` is kind of the snapshot of current span data, which can be considered as a type of `ReadableSpan`

Reply:
Yeah probably fine. It's just in the SDK anyway.

Reply:
@jtescher I have been thinking if we could have a `ReadableSpan` somehow. I guess the problem here is we cannot store an immutable reference in span processors because it will prevent mutable borrow in the future.